test/Makefile

```diff
@@ -86,3 +87,87 @@ sotest/libsotest.so.1.0: sotest/libsotest.c
 	gcc -o $@ -shared -fPIC -Wl,-soname,libsotest.so.1 -lc $^
 	ln -f -s libsotest.so.1.0 sotest/libsotest.so
 	ln -f -s libsotest.so.1.0 sotest/libsotest.so.1
+
+# PREFIX is the prefix expected at runtime (usually /usr or /usr/local for
+#  system-wide installations).
+#  More: https://www.gnu.org/prep/standards/html_node/Directory-Variables.html
+#
+# DESTDIR is the installation directory using during make install, which
+#  usually coincides for manual installation, but is chosen to be a temporary
+#  directory in packaging environments. PREFIX needs to be appended.
+#  More: https://www.gnu.org/prep/standards/html_node/DESTDIR.html
+#
+# Reasoning here: Users performing manual install *have* to specify PREFIX;
+# default is to use that also for DESTDIR. If DESTDIR is provided in addition,
+# we use that for installation.
+#
+# PREFIX can be relative unless DESTDIR is set. Absolute paths are not
+# canonicalized.
+ifneq ($(PREFIX),)
+ifneq ($(shell echo "$(PREFIX)" | cut -c1),/)
+  ifdef DESTDIR
+    $(error PREFIX must be absolute if DESTDIR is set)
+  endif
+  override PREFIX := $(abspath $(PREFIX))
+  $(warning Relative PREFIX converted to $(PREFIX))
+endif
+endif
+VERSION := $(shell cat ../VERSION.full)
+INSTALL_PREFIX := $(if $(DESTDIR),$(DESTDIR)/$(PREFIX),$(PREFIX))
+# LIBEXEC_DIR is modeled after FHS 3.0 and
+# # https://www.gnu.org/prep/standards/html_node/Directory-Variables.html. It
+# # contains any executable helpers that are not needed in PATH. Default is
+# # libexec/charliecloud which will be preprended with the PREFIX.
+LIBEXEC_DIR	?= libexec/charliecloud-$(VERSION)
+LIBEXEC_INST 	:= $(INSTALL_PREFIX)/$(LIBEXEC_DIR)
+LIBEXEC_RUN 	:= $(PREFIX)/$(LIBEXEC_DIR)
+EXAMPLES 	:= $(LIBEXEC_INST)/examples
+TEST            := $(LIBEXEC_INST)/test
```

Comment:
Most of these variables should be passed in from the parent Make rather than re-computed. The manual section on "recursive make" may be helpful on this common pattern.